pkg/provider/resource_project.go

```diff
@@ -80,73 +98,140 @@ func (p *ProjectResource) Schema() *schema.Resource {
+}
+
+type cloudProvider struct {
+	CloudProviderId   string `tfsdk:"cloud_provider_id"`
+	CloudProviderName string `tfsdk:"cloud_provider_name"`
+}
 
-	projectId, err := client.Create(ctx, project_name)
+type Project struct {
+	ID             types.String    `tfsdk:"id"`
+	ProjectID      types.String    `tfsdk:"project_id"`
+	ProjectName    types.String    `tfsdk:"project_name"`
+	UserCount      types.Int64     `tfsdk:"user_count"`
+	ClusterCount   types.Int64     `tfsdk:"cluster_count"`
+	CloudProviders []cloudProvider `tfsdk:"cloud_providers"`
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (p projectResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var config Project
+	diags := req.Config.Get(ctx, &config)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
 
+	projectId, err := p.client.Create(ctx, config.ProjectName.ValueString())
 	if err != nil {
-		return fromBigAnimalErr(err)
+		resp.Diagnostics.AddError("Error creating project", "Could not create project, unexpected error: "+err.Error())
+		return
 	}
 
-	d.SetId(projectId)
+	project, err := p.client.Read(ctx, projectId)
+	if err != nil {
+		resp.Diagnostics.AddError("Error reading project", "Could not read project, unexpected error: "+err.Error())
+		return
+	}
 
-	if err := p.read(ctx, d, meta); err != nil {
-		return diag.FromErr(err)
+	config.ID = types.StringValue(project.ProjectId)
+	config.ProjectID = types.StringValue(project.ProjectId)
+	config.ProjectName = types.StringValue(project.ProjectName)
+	config.UserCount = types.Int64Value(int64(project.UserCount))
+	config.ClusterCount = types.Int64Value(int64(project.ClusterCount))
+	config.CloudProviders = []cloudProvider{}
+
+	diags = resp.State.Set(ctx, &config)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
 	}
-	return diag.Diagnostics{}
 }
 
-func (p *ProjectResource) Read(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
-	if err := p.read(ctx, d, meta); err != nil {
-		return fromBigAnimalErr(err)
+// Read refreshes the Terraform state with the latest data.
+func (p projectResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state Project
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	project, err := p.client.Read(ctx, state.ID.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Error reading project", "Could not read project, unexpected error: "+err.Error())
+		return
+	}
+
+	state.ProjectName = types.StringValue(project.ProjectName)
+	state.UserCount = types.Int64Value(int64(project.UserCount))
+	state.ClusterCount = types.Int64Value(int64(project.ClusterCount))
+	if cps := project.CloudProviders; cps != nil {
+		for _, provider := range cps {
+			state.CloudProviders = append(state.CloudProviders, cloudProvider{
```

Comment:
project.CloudProviders is of type `Set` (https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/types/basetypes#SetType), not `List` because they're an unordered collection of unique objects.
so, plainly appending the read values will not work. 

The Read function should read the project.CloudProviders values and if they do not exist in `state.CloudProviders`, they should be added to the set.